common/lib/capa/capa/capa_problem.py

```diff
@@ -855,17 +870,67 @@ def _preprocess_problem(self, tree):  # private
                 id=response_id_str
             )
 
-            # assign one answer_id for each input type or solution type
+            # assign one answer_id for each input type
             for entry in inputfields:
                 entry.attrib['response_id'] = str(response_id)
                 entry.attrib['answer_id'] = str(answer_id)
                 entry.attrib['id'] = "%s_%i_%i" % (self.problem_id, response_id, answer_id)
                 answer_id = answer_id + 1
 
-            # Find the label and save it for html transformation step
-            responsetype_label = response.find('label')
-            problem_data[self.problem_id + '_' + str(response_id)] = {
-                'label': responsetype_label.text if responsetype_label is not None else ''
+            question_id = self.problem_id + '_' + str(response_id)
+            label = ''
+            element_to_be_deleted = None
+
+            # Extract label value from <label> tag or label attribute from inside the responsetype
+            # Extract label value from <label> tag
+            responsetype_label_tag = response.find('label')
+            if responsetype_label_tag is not None:
+                label = responsetype_label_tag.text
+                # Delete <label> tag containing question text otherwise question will be rendered twice
+                element_to_be_deleted = responsetype_label_tag
+            elif 'label' in inputfields[0].attrib:
+                # Extract label value from label attribute
+                # This is the case when we have a problem with multiple questions having OLX only
+                label = inputfields[0].attrib['label']
+                # Delete <p> tag containing question text otherwise question will be rendered twice
+                p_tag = response.xpath("preceding-sibling::p[text()='{}']".format(label))
+                if p_tag:
+                    element_to_be_deleted = p_tag[0]
+            else:
+                # neither <label> tag nor label attribute is present inside responsetype
+                # existing problem with multi-questions without --- having markdown
```

Comment:
@cahrens This can happen whether the problem is new or existing if the problem has multiple-questions not separated according to new format. I will update the comments.